aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java

```diff
@@ -76,30 +79,38 @@ public class TestS3RestSigner {
   public MinioContainer minioContainer =
       new MinioContainer(CREDENTIALS_PROVIDER.resolveCredentials());
 
-  @AfterClass
-  public static void afterClass() throws Exception {
-    if (null != httpServer) {
-      httpServer.stop();
-    }
-  }
-
-  @Before
-  public void before() throws Exception {
+  @BeforeClass
```

Comment:
I have restructured the test slightly so that the same Signer instance is used for all tests. When the scheduler existed in the code, it clearly showed the resource leak that was happening when tokens were being scheduled for refresh